core/primitives-core/src/account.rs

```diff
@@ -5,27 +5,30 @@ use crate::hash::CryptoHash;
 use crate::serialize::{option_u128_dec_format, u128_dec_format_compatible};
 use crate::types::{AccountId, Balance, Nonce, StorageUsage};
 #[cfg(feature = "protocol_feature_add_account_versions")]
-use borsh::maybestd::io::Error;
+use std::io;
 
-#[cfg(not(feature = "protocol_feature_add_account_versions"))]
-/// Per account information stored in the state.
-#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
-pub struct Account {
-    /// The total not locked tokens.
-    #[serde(with = "u128_dec_format_compatible")]
-    amount: Balance,
-    /// The amount locked due to staking.
-    #[serde(with = "u128_dec_format_compatible")]
-    locked: Balance,
-    /// Hash of the code stored in the storage for this account.
-    code_hash: CryptoHash,
-    /// Storage used by the given account, includes account id, this struct, access keys and other data.
-    storage_usage: StorageUsage,
+#[cfg(feature = "protocol_feature_add_account_versions")]
+#[derive(
+    BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Copy,
+)]
+pub enum AccountVersion {
+    V1,
 }
 
 #[cfg(feature = "protocol_feature_add_account_versions")]
-#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
-pub struct AccountV1 {
+impl Default for AccountVersion {
+    fn default() -> Self {
+        AccountVersion::V1
+    }
+}
+
+/// Per account information stored in the state.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+#[cfg_attr(
+    not(feature = "protocol_feature_add_account_versions"),
+    derive(BorshSerialize, BorshDeserialize)
```

Comment:
probably better to remove derive altogether to prevent accidental change to state

Reply:
Can you elaborate? Derive only kicks in if feature is not enabled

Reply:
Also this is covered by tests